### PR TITLE
Use `CloseIssue` and `ReopenIssue` instead of `ChangeStatus`

### PR DESCRIPTION
**Pull request by lunny**
The behaviors of closing issues and reopening issues are very different. So splitting it into two different functions makes it easier to maintain.

- [x] Split ChangeIssueStatus into CloseIssue and ReopenIssue both at service layer and model layer
- [x] Rename `isClosed` to `CloseOrReopen` to make it more readable.
- [x] Add transactions for ReopenIssue and CloseIssue

### CODE REVIEW

File: routers/api/v1/repo/issue.go [hunk 1]
```diff
@@ -912,24 +912,29 @@ func EditIssue(ctx *context.APIContext) {
 			}
 		}
 
-		var isClosed bool
+		var closeOrReopen bool
```
Comment (Zettat123) [inline routers/api/v1/repo/issue.go]: I think this variable name `closeOrReopen` is a bit weird, but I have no idea how to improve it. 🤔

File: routers/web/repo/issue_list.go [hunk 2]
```diff
@@ -469,6 +471,11 @@ func UpdateIssueStatus(ctx *context.Context) {
 				ctx.ServerError("ChangeStatus", err)
```
Comment (Zettat123) [inline routers/web/repo/issue_list.go]:
```suggestion
				ctx.ServerError("CloseIssue", err)
```

File: models/issues/issue_update.go [hunk 3]
```diff
@@ -119,16 +119,54 @@ func doChangeIssueStatus(ctx context.Context, issue *Issue, doer *user_model.Use
...
+	if err != nil {
+		return nil, err
+	}
+	if err := committer.Commit(); err != nil {
+		return nil, err
+	}
+	return comment, nil
+}
+
 // ChangeIssueStatus changes issue status to open or closed.
```
Comment (Zettat123) [inline models/issues/issue_update.go]:
```suggestion
// ReopenIssue changes issue status to open.
```

File: services/issue/status.go [hunk 4]
```diff
@@ -6,34 +6,54 @@ package issue
...
 	notify_service "code.gitea.io/gitea/services/notify"
 )
 
-// ChangeStatus changes issue status to open or closed.
-// closed means the target status
-// Fix me: you should check whether the current issue status is same to the target status before call this function
-// as in function changeIssueStatus we will return WasClosedError, even the issue status and target status are both open
-func ChangeStatus(ctx context.Context, issue *issues_model.Issue, doer *user_model.User, commitID string, closed bool) error {
-	comment, err := issues_model.ChangeIssueStatus(ctx, issue, doer, closed)
+// CloseIssue close and issue.
```
Comment (Zettat123) [inline services/issue/status.go]:
```suggestion
// CloseIssue close an issue.
```

File: models/issues/dependency_test.go [hunk 5]
```diff
@@ -49,7 +49,7 @@ func TestCreateIssueDependency(t *testing.T) {
 	assert.False(t, left)
 
 	// Close #2 and check again
-	_, err = issues_model.ChangeIssueStatus(db.DefaultContext, issue2, user1, true)
+	_, err = issues_model.CloseIssue(db.DefaultContext, issue2, user1)
```
Comment (Zettat123) [inline models/issues/dependency_test.go:52]: Looks like we only test `issues_model.CloseIssue`. Do we need any tests on `issues_model.ReopenIssue` ?

Comment (lunny) [reply models/issues/dependency_test.go:52]: a869ae7088

**Review by Zettat123 — APPROVED** [review]
Overall LGTM except CI error

**Comment by lunny** [comment]
Last call for one more review. @go-gitea/technical-oversight-committee

**Review by hiifong — APPROVED** [review]

**Pull request merged**